include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -91,27 +85,42 @@ class SubAllocatedDescriptorSet : public core::IReferenceCounted
 	uint32_t getLayoutBindingCount() { return m_allocatableRanges.size(); }
 
 	// whether that binding index can be sub-allocated
-	bool isBindingAllocatable(uint32_t binding) { return m_allocatableRanges[binding].reservedSize != 0; }
+	bool isBindingAllocatable(uint32_t binding) 
+	{ 
+		return m_allocatableRanges.find(binding) != m_allocatableRanges.end(); 
+	}
 
 	AddressAllocator& getBindingAllocator(uint32_t binding) 
 	{ 
-		assert(isBindingAllocatable(binding)); // Check if this binding has an allocator
-		return *m_allocatableRanges[binding].addressAllocator; 
+		auto range = m_allocatableRanges.find(binding);
+		assert(range != m_allocatableRanges.end());// Check if this binding has an allocator
```

Comment:
you might want to change the signature of the method to return a poitner instead of a reference so you can gracefully handle failures